src/main/kotlin/uk/gov/justice/digital/hmpps/hmppshdcapi/licences/LicenceService.kt

```diff
@@ -31,15 +26,13 @@ class LicenceService(
     val curfew = licenceData.curfew
     val proposedAddress = licenceData.proposedAddress
 
-    return curfew?.curfewHours?.let { formatCurfewHoursObject(curfew.curfewHours) }?.let {
-      HdcLicence(
-        licenceId = licence.id,
-        curfewAddress = getAddress(curfew, cas2Referral, proposedAddress),
-        firstNightCurfewHours = curfew.firstNight,
-        // curfewHours referred to as curfewTimes in CVL as going forward a more suitable name and had to distinguish between the two different curfew data formats
-        curfewTimes = it,
-      )
-    }
+    return HdcLicence(
+      licenceId = licence.id,
+      curfewAddress = transformToModelCurfewAddress(getAddress(curfew, cas2Referral, proposedAddress)),
+      firstNightCurfewHours = transformToModelFirstNight(curfew?.firstNight),
```

Comment:
generally would avoid passing nullables into functions, where they are the only parameter. to not pass in nullable and instead only transform if not null:
`curfew?.firstNight?.let{ transformToModelFirstNight(it) }`

In this case it'd mean that `firstNightCurfewHours` would be a null reference rather than an object with null times in it